package.json

```diff
@@ -10,24 +10,32 @@
   },
   "author": "Viktor Qvarfordt",
   "main": "./dist/cjs/index.js",
-  "module": "./dist/esm/index.js",
-  "types": "./dist/types/index.d.ts",
+  "module": "./dist/esm/index.mjs",
+  "types": "./dist/cjs/index.d.ts",
   "exports": {
-    "import": "./dist/esm/index.js",
-    "require": "./dist/cjs/index.js",
-    "default": "./dist/esm/index.js",
-    "types": "./dist/types/index.d.ts"
+    ".": {
+      "import": {
+        "types": "./dist/esm/index.d.mts",
+        "default": "./dist/esm/index.mjs"
+      },
+      "require": {
+        "types": "./dist/cjs/index.d.ts",
+        "default": "./dist/cjs/index.js"
+      }
+    }
   },
   "files": [
     "src/**/*",
     "dist/**/*.js",
+    "dist/**/*.mjs",
     "dist/**/*.js.map",
     "dist/**/*.d.ts",
+    "dist/**/*.d.mts",
     "dist/**/*.d.ts.map"
   ],
   "scripts": {
     "clean": "rm -rf dist",
-    "build": "tsc & tsc --project tsconfig.cjs.json",
+    "build": "yarn clean && tsc && tsc --project tsconfig.cjs.json && find dist/esm/ -name \"*.js\" -execdir bash -c 'mv \"$1\" \"${1%.js}.mjs\"' - '{}' \\; && mv ./dist/esm/index.d.ts ./dist/esm/index.d.mts",
```

Comment:
Can this be simplified? It would be good to not have do these renames

Reply:
Unfortunately it seems that `tsc` can't be configured to use a different extension other than `js` or `jsx`. You could try going with another bundler like `rollup` or `microbundle`.

Reply:
Will it work if we instead change
```
      "import": {
        "types": "./dist/esm/index.d.mts",
        "default": "./dist/esm/index.mjs"
```
to
```
      "import": {
        "types": "./dist/esm/index.d.ts",
        "default": "./dist/esm/index.js"
```
?

Reply:
No, because by default Node expects that any file is a CommonJS module. You can change this by including `"type": "module"` in the `package.json` so every file will be treated as a ES module.

The problem is that you can only have one type of module by default, and you must mark the others as different by using an explicit extension. So right now all your `.js` files are expected to be CommonJS modules. Any ES module must use one of the `mjs` or `d.mts` extensions to be treated as such.

Reply:
Makes sense. Thanks. Let’s merge!